### PR TITLE
MGMT-2380 Update README

### PR DESCRIPTION
**Pull request by empovit**

### CODE REVIEW

**Comment by empovit** [comment]
IMO it would be easier just to open the file and read through it, instead of looking at the diff.

File: README.md [hunk 1]
```diff
@@ -1,58 +1,61 @@
...
+              [assisted installer](https://github.com/openshift/assisted-service) and starts **next_step_runner**.
+* **next_step_runner** - Polls the assisted installer service for next steps of an assisted installation, and executes them.
+* **inventory** - Provides information about host's inventory (CPU, memory, disk, network interfaces, etc.).
+* **connectivity_check** - Tests connectivity via NICs in a list provided as the argument.
+* **free_addresses** - Detects free addresses in a subnet provided as the argument.
+* **logs_sender** - Packages system logs and uploads them to the server for troubleshooting.
+* **dhcp_lease_allocate** - Allocates IP addresses in DHCP. An assisted cluster needs several reserved IPs such as API VIP and ingress VIP.
+* **apivip_check** - Tests connectivitiy to the API VIP of the assisted cluster.
+
+The agent uses the **ghw** library to generate unique uuid identifying the host that the agent is running on.
```
Comment (YuviGold) [inline README.md:16]: It has a lot more libraries. What does it matter to specify this certain lib?

**Review by YuviGold — CHANGES_REQUESTED** [review]

File: README.md [hunk 2]
```diff
@@ -1,58 +1,61 @@
...
 ### Packaging
-The agent's executable is packaged in a container quay.io/ocpmetal/agent:latest.  The excutable in the container is /usr/bin/agent.
+The agent's executable is packaged in a container image `quay.io/ocpmetal/assisted-installer-agent:latest`.
+The excutable inside the image is `/usr/bin/agent`.
 
 ### Running
-Since the agent is a statically linked go executable it can be copied and ran outside a container.If running outside a container, 
-the agent shoud run as root.  If running in a container, the docker or podman should be invoked with --net=host and --privileged flags.
+Since the agent is a statically linked go executable it can be copied and ran outside a container. If running outside a container,
+the agent shoud run as root. If running in a container, the docker or podman should be invoked with `--net=host` and `--privileged` flags.
```
[the remark this answers is not in the document]
Comment (empovit) [reply README.md:37]: Right, will change that.

File: README.md [hunk 3]
```diff
@@ -1,58 +1,61 @@
...
 
 ### Dependecies
-* docker
+* Docker
```
[the remark this answers is not in the document]
Comment (empovit) [reply README.md:40]: Will add

File: README.md [hunk 4]
```diff
@@ -1,58 +1,61 @@
...
+### Agent Flags
+
+* *--url*: The inventory URL, including a schema and optionally a port.
+* *--cluster-id*: Cluster ID the host will be connected to.
+* *--agent-version*: Version (image tag) of the agent being run, collected for diagnostic purposes.
 * *--interval*: Interval in seconds between consecutive times that the agent accesses the inventory.  Default is 60.
-* *--inventory-image*: The name of the inventory image.  Default is "quay.io/ocpmetal/inventory:latest"
-* *--text*: Dump the inventory as json to the standard output and exit.
-* *--connectivity*: Perform connectivity check and dump the results as json to the standard output.  The details of the nodes to check the connectivity with are provided as a parameter having json format.
+* *--inventory-image*: The name of the inventory image.  Default is "quay.io/ocpmetal/inventory:latest".
```
[the remark this answers is not in the document]
Comment (empovit) [reply README.md:24]: It _is_ inventory, but probably isn't needed. Good catch!

File: README.md (hunk 1 above)
Comment (empovit) [reply README.md:16]: Looks like it, but I guess the original author had a good reason to do so.

Comment (mhrivnak) [inline README.md:16]: A link would be valuable here.

File: README.md [hunk 5]
```diff
@@ -1,58 +1,61 @@
...
-Currently, the functionality of **connectivity_check** is also bundled in the agent itself, besides its dedicated container images.
-
-### Flags
-
-* *--host*:  **Deprecated, use *--url* instead.** The inventory host as dns name or ip address. Default is "api.openshift.com".
-* *--port*: **Deprecated, use *--url* instead.** The inventory port number. Default is 80.
-* *--url*: The inventory URL, including a schema and optionally a port. Overrides the host and port arguments. If not provided, "http://\<host\>:\<port\>" will be used.
-* *--cluster-id*: The cluster id.  Default is "default-cluster".
+
+This projects generates several products, each packaged as a Docker image:
```
Comment (mhrivnak) [inline README.md:4]: s/Docker image/container image/

File: README.md (hunk 2 above)
Comment (mhrivnak) [inline README.md:37]: s/shoud/should/

File: README.md (hunk 3 above)
Comment (mhrivnak) [inline README.md:40]: When run in an RHCOS live ISO, doesn't it use podman?

**Review by mhrivnak — CHANGES_REQUESTED** [review]

File: README.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (empovit) [reply README.md:4]: "binary"?

File: README.md (hunk 3 above)
Comment (empovit) [reply README.md:40]: Docker is needed for subsystem testing. For now at least.

Comment (mhrivnak) [reply README.md:40]: It's probably worth noting then that it is just a testing dependency, and not a runtime dependency.

File: README.md [hunk 6]
```diff
@@ -1,58 +1,70 @@
...
-* *--cluster-id*: The cluster id.  Default is "default-cluster".
-* *--interval*: Interval in seconds between consecutive times that the agent accesses the inventory.  Default is 60.
-* *--inventory-image*: The name of the inventory image.  Default is "quay.io/ocpmetal/inventory:latest"
-* *--text*: Dump the inventory as json to the standard output and exit.
-* *--connectivity*: Perform connectivity check and dump the results as json to the standard output.  The details of the nodes to check the connectivity with are provided as a parameter having json format.
-* *--with-text-logging*: Enable writing the agent logs to /var/log/agent.log. Default is true.
-* *--with-journal-logging*: Enable writing logs to systemd journal. Default is true.
-* *--insecure*: Skip certificate validation in case of HTTPS transport. Should be used only for testing. Default is false.
+
+This projects contains several executables that are intended to run in tandem with the
```
Comment (danielerez) [inline README.md:5]: hmm, nothing crucial, but just not sure about 'tandem' :)
maybe: 'intended to be invoked by...' or 'intended to run along with...'

File: README.md [hunk 7]
```diff
@@ -1,58 +1,70 @@
...
+[OpenShift Assisted Installation Service](https://github.com/openshift/assisted-service):
```
Comment (danielerez) [inline README.md:6]: iinm, it's commonly named 'installer service'

File: README.md [hunk 8]
```diff
@@ -1,58 +1,70 @@
...
+By default, the executables are packaged in a container image `quay.io/ocpmetal/assisted-installer-agent:latest`.
+The executables inside the image reside under _/usr/bin/_.
 
 ### Running
-Since the agent is a statically linked go executable it can be copied and ran outside a container.If running outside a container, 
-the agent shoud run as root.  If running in a container, the docker or podman should be invoked with --net=host and --privileged flags.
 
-### Dependecies
-* docker
+Since the agent is a statically linked go executable, it can be copied and ran outside a container. If running outside a container,
```
Comment (danielerez) [inline README.md:40]: s/ran/run

File: README.md (hunk 1 above)
Comment (empovit) [reply README.md]: Apparently the library plays an important role in inventory collection, I updated this with a link to it repo.

**Review by ori-amizur — APPROVED** [review]
/lgtm

**Pull request merged**